constants.js

```diff
@@ -40,4 +40,23 @@ export const tokensHost = `https://${chainName}all-tokens.webaverse.com`;
 export const landHost = `https://${chainName}sidechain-land.webaverse.com`;
 export const web3MainnetSidechainEndpoint = 'https://mainnetsidechain.exokit.org';
 export const web3TestnetSidechainEndpoint = 'https://testnetsidechain.exokit.org';
-export const homeScnUrl = `https://webaverse.github.io/street/street.scn`;
+export const homeScnUrl = 'https://webaverse.github.io/street/street.scn';
+
+// Setup dialog URL, use hash if on githubpreview and its an okay hash
+const rGithub = /\.githubpreview\.dev$/;
+const isGH = rGithub.test(window.location.hostname);
+
+const hashHost =
+  /^[a-z][a-z0-9+\-.]*:\/\/([a-z0-9\-._~%!$&'()*+,;=]+@)?([a-z0-9\-._~%]+|\[[a-z0-9\-._~%!$&'()*+,;=:]+\])/.exec(
+    window.location.hash.slice(1),
+  );
+
+const isOKHASH = hashHost && hashHost[2] && rGithub.test(hashHost[2]);
+
+export const dialogUrl = isGH ? 
+isOKHASH ? 
+  hashHost[2] 
+  : 
+  window.location.hostname.replace('3000', '4443')
```

Comment:
It also seems strange to assume 3000 here; there are better ways to add a port, like a window.location.host, which excludes it. Additionally any ports we use should probably be constants.

Reply:
Its part of the hostname ie: https://agoblinking-app-wr9v5wqfvr4-3000.githubpreview.dev/

Reply:
the 3000 is assumed because our index.js uses that 
![image](https://user-images.githubusercontent.com/462380/130141700-9202bab7-0672-471a-8e4d-80faf2dbda6c.png)

Reply:
location.host is with port location.hostname is w/o port.

Reply:
Oh I see, in that case it could be done with a stronger regex.

Reply:
I think the only edge case with 3000 is if someone had it in their name or project title.

 I'll narrow it to avoid those. Are there any others you're aware of?

Reply:
I just think it should operate on a whitelist basis, not a replace basis.

```
/\.[0-9]+\.githubpreview\.dev$/
```

Reply:
Or, what you have here works. 👍

Reply:
We'll have to do something fancy with ports and known URLs eventually that makes it easier to switch between dev and production.